src/common/utils.js

```diff
@@ -0,0 +1,309 @@
+const constants = require('../common/constants');
+
+class Queue {
+  /**
+   * Implements Javascript Queue.
```

Comment:
Did you implement this yourself? If not, it'd be nice to specify the reference link as an inline comment.

Reply:
yes I implemented this class here. I'll add a script to generate headers (with author and registry) for all the files once this PR gets merged :)